broker/consumer/rubin/README.md

```diff
@@ -0,0 +1,321 @@
+    --allow=tcp:9094 \
+    --description="Allow incoming traffic on TCP port 9094" \
+    --direction=INGRESS \
+    --enable-logging
+
+# Create a Cloud Storage bucket to store the consumer config files
+gsutil mb "gs://${broker_bucket}"
+
+# Upload the install script and config files for the consumer
+o="GSUtil:parallel_process_count=1" # disable multiprocessing for Macs
+gsutil -m -o "$o" cp -r broker/consumer "gs://${broker_bucket}"
+
+# Create a Pub/Sub topic and subscription for Rubin alerts
+gcloud pubsub topics create $PUBSUB_TOPIC
+gcloud pubsub subscriptions create $PUBSUB_SUBSCRIPTION --topic=$PUBSUB_TOPIC
+
+# Create a Rubin Consumer VM
+zone="us-central1-a"
+machinetype="e2-standard-2"
+installscript="gs://${broker_bucket}/consumer/vm_install.sh"
+gcloud compute instances create "$consumerVM" \
+    --zone="$zone" \
+    --machine-type="$machinetype" \
+    --scopes=cloud-platform \
+    --metadata=google-logging-enabled=true,startup-script-url="$installscript" \
+    --tags="$firewallrule"
+```
+
+# Ingest the Rubin test stream
+
+## Setup
+
+```bash
+# start the consumer vm and ssh in
+gcloud compute instances start "$consumerVM"
+gcloud compute ssh "$consumerVM"
+
+# define some variables
+brokerdir=/home/broker
+workingdir="${brokerdir}/consumer/rubin"
+# Also define the variables from the "Setup" section
+# at the very top of this README file.
+```
+
+## Test the connection
+
+### Check available Kafka topics
+
+```bash
+/bin/kafka-topics \
+    --bootstrap-server alert-stream-int.lsst.cloud:9094 \
+    --list \
+    --command-config "${workingdir}/admin.properties"
+# should see output that includes the topic: alerts-simulated
+```
+
+### Test the topic connection using the Kafka Console Consumer
+
+Set Java env variable
+
+```bash
+export JAVA_HOME="/usr/lib/jvm/java-11-openjdk-amd64"
+```
+
+Make a file called 'consumer.properties' and fill it with this
+(change `KAFKA_PASSWORD` to the appropriate value):
+
+```bash
+security.protocol=SASL_SSL
+sasl.mechanism=SCRAM-SHA-512
+
+sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required \
+  username="pittgoogle-idfint"\
+  password="KAFKA_PASSWORD";
+```
+
+Run the Kafka console consumer
+
+```bash
+sudo /bin/kafka-avro-console-consumer \
+    --bootstrap-server alert-stream-int.lsst.cloud:9094 \
+    --group "${KAFKA_USERNAME}-example-javaconsole" \
+    --topic "$KAFKA_TOPIC" \
+    --property schema.registry.url=https://alert-schemas-int.lsst.cloud \
+    --consumer.config consumer.properties \
+    --timeout-ms=60000
+# if successful, you will see a lot of JSON flood the terminal
+```
+
+## Run the Kafka -> Pub/Sub connector
+
+Setup:
+
+```bash
+# download the config files from broker_bucket
+sudo mkdir "$brokerdir"
+sudo gsutil -m cp -r "gs://${broker_bucket}/consumer" "$brokerdir"
+
+# set the password in two of the config files
+sudo sed -i "s/KAFKA_PASSWORD/${KAFKA_PASSWORD}/g" "${workingdir}/admin.properties"
```

Comment:
I believe `sudo` is necessary here because I'm operating outside of my home directory (and haven't adjusted my permissions). I `ssh`d in and manually ran these commands. If I let the machine run this as a script using its own service account it shouldn't need `sudo`.